src/message/MoveMessage.js

```diff
@@ -0,0 +1,211 @@
+/* @flow strict-local */
+import React, { useState, useContext } from 'react';
+import { Text, View, Platform, Picker, TouchableOpacity, ScrollView } from 'react-native';
+import type { Node } from 'react';
+import { ThemeContext, BRAND_COLOR } from '../styles';
+import type { RouteProp } from '../react-navigation';
+import * as api from '../api';
+import Input from '../common/Input';
+import { streamNarrow, streamIdOfNarrow } from '../utils/narrow';
+import { getStreamForId } from '../subscriptions/subscriptionSelectors';
+import type { AppNavigationProp } from '../nav/AppNavigator';
+import { getAuth, getStreams, getOwnUser } from '../selectors';
+import { useSelector } from '../react-redux';
+import { showErrorAlert, showToast } from '../utils/info';
+import { Icon } from '../common/Icons';
+import type { Narrow, Message, Outbox } from '../types';
+import TopicAutocomplete from '../autocomplete/TopicAutocomplete';
+import { TranslationContext } from '../boot/TranslationProvider';
+import ZulipButton from '../common/ZulipButton';
+
+type Props = $ReadOnly<{|
+  navigation: AppNavigationProp<'move-message'>,
+  route: RouteProp<'move-message', {| message: Message | Outbox, messageNarrow: Narrow |}>,
+|}>;
+
+const inputMarginPadding = {
+  paddingHorizontal: 8,
+  paddingVertical: Platform.select({
+    ios: 8,
+    android: 2,
+  }),
+};
+
+export default function MoveMessage(props: Props): Node {
+  const themeContext = useContext(ThemeContext);
+  const backgroundColor = themeContext.backgroundColor;
+  const cardColor = themeContext.cardColor;
+  const iconName = Platform.OS === 'android' ? 'arrow-left' : 'chevron-left';
+  const auth = useSelector(getAuth);
+  const allStreams = useSelector(getStreams);
+  const isAdmin = useSelector(getOwnUser).is_admin;
+  const messageId = props.route.params.message.id;
+  const currentStreamId = streamIdOfNarrow(props.route.params.messageNarrow);
+  const currentStreamName = useSelector(state => getStreamForId(state, currentStreamId)).name;
+  const [narrow, setNarrow] = useState(streamNarrow(currentStreamId));
+  const [subject, setSubject] = useState(props.route.params.message.subject);
```

Comment:
These and some other spots would be made a bit simpler to read by pulling the route params out as locals:
```
  const { message, messageNarrow } = props.route.params;
```

That also replaces the need for a `messageId` local -- can just say `message.id`.